test/async-hooks/test-async-local-storage-properly-contained-sync.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const { AsyncLocalStorage } = require('async_hooks');
+
+const asyncLocalStorage = new AsyncLocalStorage();
+const outer = new Map();
+const inner = new Map();
+
+asyncLocalStorage.run(outer, () => {
```

Comment:
As `run*` methods merge is now reverted, a similar test should be added for `*SyncAndReturn` methods.

Also, the name of this test (`test-async-local-storage-properly-contained-sync.js`) mentions `sync`, which is not the case with `run`.